sky/serve/load_balancing_policies.py

```diff
@@ -25,6 +29,24 @@ class LoadBalancingPolicy:
     def __init__(self) -> None:
         self.ready_replicas: List[str] = []
 
+    def __init_subclass__(cls, name: str, default: bool = False):
+        LB_POLICIES[name] = cls
+        if default:
+            global DEFAULT_LB_POLICY
+            assert DEFAULT_LB_POLICY is None, (
+                'Only one policy can be default.')
+            DEFAULT_LB_POLICY = name
+
+    @classmethod
+    def make(cls, policy_name: Optional[str] = None) -> 'LoadBalancingPolicy':
+        """Create a load balancing policy from a name."""
+        if policy_name is None:
+            policy_name = DEFAULT_LB_POLICY
+
+        if policy_name not in LB_POLICIES:
+            raise ValueError(f'Unknown load balancing policy: {policy_name}')
```

Comment:
Lets also check this in `SkyServiceSpec`, before the user launch the service. If the user has a typo in the yaml, we might want to let the user konw it before launch a controller.

https://github.com/skypilot-org/skypilot/blob/c6ae536d8dfedc3bbcf427a81480382b9d5f4c29/sky/resources.py#L810-L823

Reply:
Super good point 👍